version: 2.0.0

damencs opened the pull request:
- add: config sections
- add: consume options (sound/area)
- add: whitelist ids (sound/area)
- add: blacklist ids (sound/area)
- add: debug UI tool to review sound ids (sound/area/silent)
- add: license snippets to files
- cleanup: checkstyle
- fix: security exception log for directory creation if permissions do not allow
- update: plugin properties for plugin hub and version (2.0)

damencs committed:
version: 2.0.0 initial

- add: config sections
- add: consume options (sound/area)
- add: whitelist ids (sound/area)
- add: blacklist ids (sound/area)
- add: debug UI tool to review sound ids (sound/area/silent)
- add: license snippets to files
- cleanup: checkstyle
- fix: security exception log for directory creation if permissions do not allow
- update: plugin properties for plugin hub and version (2.0)

petertalbanese commented on src/main/java/com/soundswapper/SoundSwapperPlugin.java:
```diff
@@ -52,93 +94,255 @@ SoundSwapperConfig provideConfig(ConfigManager configManager)
+	{
+		if (configText == null || configText.isEmpty())
+		{
+			return null;
```
Here's the source of the nullptr exception. Return empty list instead of null.

I think this just fixes the issues haha. Will have to test some more to make sure.

damencs committed:
fix: config returning empty list instead of null & patch: java version to 11

- fix: `return List.of();` instead of `null`
- patch: build version of java to 1.11 instead of 1.8 to follow hub recommended java version

damencs committed:
fix: targetCompatibility to VERSION_11

damencs committed:
fix: config handling and variable resets

- fix: config updating all values even if only one changed
- fixes: variables not being reset when turning the plugin off and back on

damencs committed:
adjustments: finalizations

petertalbanese committed:
adjustments: allow sounds to overlap

- add Sound class to store byte stream directly
- change Clip maps to Sound maps
- create Clip on-demand to play sounds

petertalbanese committed:
style: adjust formatting for consistency

- change tabs to 4 spaces
- move brackets to newlines

damencs committed:
remove: allowSimultaneousSounds config item

damencs committed:
add: custom volume for sound types

damencs committed:
fix: checkstyle
